tests/python/pants_test/backend/jvm/tasks/jvm_compile/zinc/zinc_compile_integration_base.py

```diff
@@ -251,6 +251,30 @@ def test_combination(target, expect_success, extra_args=[]):
     test_combination('fatal', expect_success=False,
       extra_args=['--compile-rsc-compiler-option-sets-disabled-args={"fatal_warnings": ["-S-Xfatal-warnings"]}'])
 
+  def test_zinc_logs_warnings_properly(self):
```

Comment:
Given the pretty clear description of the failings of the `BareBonesLogger` in #8279, I really want to make sure we don't regress further. To that end, it seems really easy to add a test so that we can at least make sure the `BareBonesLogger` we're hiding behind a flag can even run correctly. To avoid asking you to do too much, I've made this into a diff which you should be able to cleanly apply to this one: https://github.com/blorente/pants/compare/blorente/hide-barebones-logger-behind-flag...cosmicexplorer:blorente/hide-barebones-logger-behind-flag.

Reply:
I think this change is good but it should be done in a follow-up PR, because using the `-use-barebones-logger` flag requires this PR to have landed and a new version of the compiler being published to maven.

Otherwise, this test will fail until we bump the zinc version.

Reply:
Opened https://github.com/pantsbuild/pants/issues/8306 to track and added a TODO linking to it.

Reply:
yep, sounds good!

Reply:
+1